### ISSUE TITLE
USGS data download and identifier sorting request

### ISSUE THREAD

**Issue by jldunnum** [ArctosDB/arctos#8042]
How can I download the specimen records in this project (https://arctos.database.museum/search.cfm?project_id=10002364) using the advanced_bio_geo profile and then get the identifiers into separate columns so we can sort by "United States Geological Survey identifier" and display the BS/FC number in its own column?
I tried doing an identifier download and merge but it crashed two different computers so far. Plus I don't want to lose the other identifers if I filter specifically for "United States Geological Survey Identifier".
Used to be able to simply select this identifier in my profile and all was great in seconds. Now I spend hours trying to do work arounds and still fail........
USGS Deputy Regional Director is arriving tomorrow morning and wants to see their data and learn how to navigate Arctos and I can't even do it myself anymore much less teach her.
Any assistance would be most appreciated.

**Comment by dustymc** [comment]
I can't do anything in the UI by tomorrow, and I can't do much in the UI as long as there are a bunch of denormalizing types hanging around. There's a fairly thorough demo of what can be done if we can get past the blocks in https://github.com/ArctosDB/arctos/issues/6983 - it basically boils down to "select this identifier in my profile and all was great in seconds" but sorta accidentally avoids a bunch of ways that these data become scattered and greatly simplifies everything in the process. Maybe we can get together with @mkoo  and sort that out soonish?

Anyway, data below. I added a "usgs" column to https://arctos.database.museum/search.cfm?project_id=10002364&sp=advanced_bio_geo, results attached, let me know if I can do something else.

[temp_usgs.csv.zip](https://github.com/user-attachments/files/16754195/temp_usgs.csv.zip)

**Comment by mkoo** [comment]
@jldunnum what are the fields you would ideally like to see? (please list them all: GUID, sci_name, USGS identifer and value, verbatim date, spec_localties, dec_lat, dec_long, datum etc etc)

@dustymc can you provide a SQL to retrieve that data in this project? I think this would be a solid use case for a project report, but meanwhile we can use SQL as a workaround for now to generate a spreadsheet

does that work?

**Comment by jldunnum** [comment]
Thanks Dusty and Michelle, I can only access from my phone at the moment so can’t go in too deep easily at the moment. I’ll check out the links when I get back to a computer. Thanks, Jon

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jldunnum** [comment]
That download worked well for the USGS visit Dusty, thanks. 
Can you do another one for this project (https://arctos.database.museum/search.cfm?customoidoper=LIST&project_id=10002364) with the fields below, the USGS field you previously added, plus fields for entered and edited dates?

use_license_url | collection_object_id | guid | scientific_name | country | state_prov | spec_locality | verbatim_date | dec_lat | dec_long | coordinateuncertaintyinmeters | sex | parts | age_class | othercatalognumbers | phylorder | family | county | nk | identified_by | preparators | remark | accn_number | began_date | minimum_elevation | orig_elev_units | collectors | datum | entered_by | ear_from_notch | hind_foot_with_claw | tail_length | total_length | weight | collector_number | genbank | preparator_number | reproductive_data

Thanks much, Jon

**Comment by dustymc** [comment]
>with the fields below

Pretty please: click your link, pick a profile eg...

<img width="327" alt="Screenshot 2024-08-27 at 11 46 59" src="https://github.com/user-attachments/assets/26992026-ebdd-4970-b393-981232c8848b">


then...

<img width="273" alt="Screenshot 2024-08-27 at 11 47 47" src="https://github.com/user-attachments/assets/d027ed35-abd5-4f3c-89c0-186c3d6ea3cd">

and send me that URL.

**Comment by jldunnum** [comment]
Here ya go
https://arctos.database.museum/search.cfm?customoidoper=LIST&project_id=10002364
______________________________________________________________
Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
Research Assistant Professor (LAT)
Department of Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by dustymc** [comment]
That isn't it, it should have a "sp={your profile name}" in there somewhere.

**Comment by jldunnum** [comment]
https://arctos.database.museum/search.cfm?customoidoper=LIST&project_id=10002364&sp=Dunnum_USGS

**Comment by dustymc** [comment]
Here you go:
[temp_usgs.zip](https://github.com/user-attachments/files/16768648/temp_usgs.zip)

And because I noticed one:

```
create table temp_dups as select usgs,count(*) c from temp_usgs group by usgs having count(*) > 1;
```
[temp_dups.csv.zip](https://github.com/user-attachments/files/16768659/temp_dups.csv.zip)

**Comment by jldunnum** [comment]
Yes, one unfortunate issue is that every taxonomic division within the USGS collection has the same numbering scheme so there could be three or four BS/FC 1.....



______________________________________________________________
Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
Research Assistant Professor (LAT)
Department of Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by dustymc** [comment]
done?

**Comment by campmlc** [comment]
Is there SQL we can use to get these data? Or to parse any identifier into its own column? Or possible to create a profile with this info?

**Comment by campmlc** [comment]
@dusty - please see questions above.

**Comment by mkoo** [comment]
Closing this issue (again) since the original request is complete. See this #8091

**Comment by jldunnum** [comment]
Adding to this thread because todays issue is basically the same type of problem/request. Spent last week at the AMNH and they are eager to have a download of our shared data to update their database and sort out some data problems they have. The MSB worked on a decade long collaborative project with AMNH in the 1980s and 90s in Bolivia. AMNH didn't have a tissue collection at that time so all tissues came to MSB. Thus we have large series of specimens cataloged here which are tissue only and correspond to skins/skels/fluids held at AMNH. These all have NK numbers, MSB numbers and AMNH numbers. I'd like to have a parsed file with MSB guid, AMNH institutional catalog number, NK number, collector name and number, preparator name and number, higher geography, specific locality, georeference data, sex, date. 
Bottom line is this type of request is going to keep coming up so I really need the ability to use one of my profiles to download records and then parse out all the identifiers into their own columns. Thanks
